core/classes/Misc/IntegrityChecker.php

```diff
@@ -0,0 +1,147 @@
+<?php
+
+class IntegrityChecker {
+
+    const CHECKSUMS_PATH = ROOT_PATH . '/checksums.json';
+
+    /**
+     * Files with relative paths starting with a string in this array are ignored
+     */
+    const IGNORED_PATHS = [
+        'checksums.json',
+        'cache/',
+        'cache/logs/',
+        'cache/templates_c/',
+        'templates/', # The default template is included again, below
+        'modules/', # Default modules are included again, below
+        'uploads/',
+    ];
+
+    /**
+     * Override a path within an ignored path to be included in the scan
+     */
+    // TODO: Have constants for default template and module names somewhere, so it doesn't have to be hardcoded here (and in other code)
+    const INCLUDED_PATHS = [
+        'cache/.htaccess',
+        'templates/DefaultRevamp',
+        'modules/Cookie Consent',
+        'modules/Core',
+        'modules/Discord Integration',
+        'modules/Forum',
+    ];
+
+    /**
+     * @return bool If the file should be ignored from integrity checking, according to IGNORED_PATHS and INCLUDED_PATHS.
+     */
+    private static function is_ignored($path): bool {
```

Comment:
Can we swap these to camel case; isIgnored?

Reply:
right, I forgot functions where snake case but methods camel case